pkg/endpoint/endpoint.go

```diff
@@ -2606,9 +2613,11 @@ func (e *Endpoint) syncPolicyMapController() {
 	ctrlName := fmt.Sprintf("sync-policymap-%d", e.ID)
 	e.controllers.UpdateController(ctrlName,
 		controller.ControllerParams{
-			DoFunc: func() error {
-				e.Mutex.Lock()
-				defer e.Mutex.Unlock()
+			DoFunc: func() (reterr error) {
+				if lockerr := e.LockAlive(); lockerr != nil {
```

Comment:
This will help a lot as well! I don't think we have a return an error here because the endpoint disappearing means the regular end of the controller.

Reply:
We can add a debug message if we want but it's not an error condition.

Reply:
Added a debug message and returning no error here then.